common/src/main/java/net/caffeinemc/mods/sodium/mixin/features/textures/animations/tracking/SpriteContentsTickerMixin.java

```diff
@@ -57,4 +74,28 @@ private void postTick(CallbackInfo ci) {
         SpriteContentsExtension parent = (SpriteContentsExtension) this.parent;
         parent.sodium$setActive(false);
     }
+
+    @Override
+    public void sodium$ensureUpload(ResourceLocation atlas, int x, int y) {
+        if (!this.skippedUpload) {
+            return;
+        }
+
+        Minecraft.getInstance().getTextureManager().getTexture(atlas).bind();
```

Comment:
Revisiting this, its probably problematic to change the bound texture here since we can't know at what point in the code a sprite is being marked active and it might mess up render state.

Reply:
This issue is not present with just the first commit, but as stated in the initial pr message the behaviour from just the first commit is not 100% correct since it leaves a delay between the texture becoming visible and when its reticked